.github/workflows/test-deptrac.yml

```diff
@@ -0,0 +1,82 @@
+# When a PR is opened or a push is made, perform an
+# architectural inspection on the code using Deptrac.
+name: Deptrac
+
+on:
+  pull_request:
+    branches:
+      - 'develop'
+      - '4.*'
+    paths:
+      - 'app/**'
+      - 'system/**'
+      - 'composer.json'
+      - 'depfile.yaml'
+      - '.github/workflows/test-deptrac.yml'
+  push:
+    branches:
+      - 'develop'
+      - '4.*'
+    paths:
+      - 'app/**'
+      - 'system/**'
+      - 'composer.json'
+      - 'depfile.yaml'
+      - '.github/workflows/test-deptrac.yml'
+
+jobs:
+  build:
+    name: PHP ${{ matrix.php-versions }} Architectural Inspection
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.4', '8.0']
```

Comment:
Is there merit for testing it on different PHP versions? I mean we can test only on highest version allowed.

Reply:
Hmm that's a good question. I'm not sure? Probably we don't need both, I can't imagine they would amount to meaningful differences. Most of this workflow file was cobbled together from existing templates, so I won't say every part of it was intentional :D

Reply:
I would go for one version only since this only tests architectural relations which I believe won't change in between PHP versions.